### PR TITLE
Remove live PV during inference

### PR DESCRIPTION
**Pull request by zakwatts**
A --no-live-pv flag has been added to eval script so that when used, features generated from live pv from a site are set to NaNs as well as the appropriate is NaN labelling feature set to 1 (which is what the model expects for NaNs).

Experiments have been performed looking into the performance of a bespoke vs generic model as well as the effect of the amount of pv dropout used during training. 

A small change has been made to the training script. To avoid inference --no_infer flag was used however this has now been changed to --no-infer.

### CODE REVIEW

File: psp/exp_configs/uk_pv_site_comp.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import datetime as dt
```
Comment (peterdudfield) [inline psp/exp_configs/uk_pv_site_comp.py:1]: Could you add a desrciption at the top, here, im not sure what `comp` or `comp_gen` means?

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Ive approved it but would be great to get `lint` working first

**Comment by peterdudfield** [comment]
Also would be good to put results in exp folder once you have them

**Comment by zakwatts** [comment]
Yes, ill add a exp_report tomorrow. I'll keep this PR open for now and merge once I have completed it.

**Comment by peterdudfield** [comment]
> Yes, ill add a exp_report tomorrow. I'll keep this PR open for now and merge once I have completed it.

nice, ive made the PR a draft, so then we'll know when its ready.

**Comment by zakwatts** [comment]
@peterdudfield instead of --use-metadata, i think --no-live-pv makes more sense?

File: psp/models/multi.py [hunk 2]
```diff
@@ -25,6 +27,23 @@ def get_features(self, x: X, is_training: bool = False) -> Features:
         model = self._get_model_for_ts(x.ts)
         return model.get_features(x, is_training=is_training)
 
+    def get_features_without_pv(self, x: X, is_training: bool = False) -> Features:
+        features = self.get_features(x, is_training=is_training)
```
Comment (peterdudfield) [inline psp/models/multi.py:31]: add function description

File: exp_reports/014_remove_live_PV_at_inference/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Generic vs Bespoke Models With and Without Live PV
+
```
Comment (peterdudfield) [inline exp_reports/014_remove_live_PV_at_inference/README.md:2]: add small summary

File: exp_reports/014_remove_live_PV_at_inference/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Training using a PV dropout of 0.1, 0.2 and 0.3 have been explored to see if there are any significant changes in the error.
+
+![results](pv_dropout_change.png)
+*Figure 2: Change in error relative to where pv dropout is equal to 0.1, training with 4096 samples.*
+
+There is a very small change in error. Changing the dropout to 0.2 (around 800 samples) has a slight increase in the error for the case without live pv and slight decrease in error for the case with live pv. With a pv droupout of 0.3 both cases see a decrease in error.
+
+# Effects of using household model on larger sites
+
```
Comment (peterdudfield) [inline exp_reports/014_remove_live_PV_at_inference/README.md:62]: do in different exp

File: psp/models/multi.py [hunk 5]
```diff
@@ -28,6 +28,18 @@ def get_features(self, x: X, is_training: bool = False) -> Features:
...
+        It first calls the `get_features` method to generate all features, then sets the value of PV-derived features to NaN and the 
+        value of NaN labelling features to 1.
+    
+        Args:
+            x (X): The input data to generate features from.
+            is_training (bool, optional): A flag indicating whether the function is being called during training. Defaults to False.
+    
+        Returns:
+            features: A dictionary of features, with PV-derived features set to NaN and NaN labelling features set to 1.
+        """
```
Comment (peterdudfield) [inline psp/models/multi.py:42]: very nice
